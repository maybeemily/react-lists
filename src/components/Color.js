import React from 'react';
import PropTypes from 'prop-types';

function Color({ name, hex, rgb }) {
  return (
    <dl>
      <dt>Name</dt>
      <dd><div style={{ display: 'inline-block', backgroundColor: hex, width: '20px', height: '20px' }}></div>{name}</dd>

      <dt>Hex</dt>
      <dd>{hex}</dd>

      <dt>RBG</dt>
      <dd>{rgb}</dd>
    </dl>
  );
}

Color.propTypes = {
  name: PropTypes.string.isRequired,
  hex: PropTypes.string.isRequired,
  rgb: PropTypes.string.isRequired
};

export default Color;
