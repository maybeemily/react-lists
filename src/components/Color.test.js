import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color component', () => {
  it('renders a color', () => {
    const testColor = { name: 'hotpink', hex: '#FF69B4', rgb: '(255,105,180)' };
    const wrapper = shallow(<Color name={testColor.name} hex={testColor.hex} rgb={testColor.rgb} />);
    expect(wrapper).toMatchSnapshot();
  });
});
