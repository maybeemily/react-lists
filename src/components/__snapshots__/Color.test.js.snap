// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component renders a color 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    <div
      style={
        Object {
          "backgroundColor": "#FF69B4",
          "display": "inline-block",
          "height": "20px",
          "width": "20px",
        }
      }
    />
    hotpink
  </dd>
  <dt>
    Hex
  </dt>
  <dd>
    #FF69B4
  </dd>
  <dt>
    RBG
  </dt>
  <dd>
    (255,105,180)
  </dd>
</dl>
`;
